feg/gateway/services/n7_proxy/servicers/session_controller.go

```diff
@@ -0,0 +1,211 @@
+func (srv *CentralSessionController) CreateSession(
+	ctx context.Context,
+	request *protos.CreateSessionRequest,
+) (*protos.CreateSessionResponse, error) {
+
+	return nil, nil
+}
+
+// UpdateSession handles periodic updates from gateways that include quota
+// exhaustion and terminations.
+func (srv *CentralSessionController) UpdateSession(
+	ctx context.Context,
+	request *protos.UpdateSessionRequest,
+) (*protos.UpdateSessionResponse, error) {
+
+	return nil, nil
+}
+
+// TerminateSession handles a session termination
+func (srv *CentralSessionController) TerminateSession(
+	ctx context.Context,
+	request *protos.SessionTerminateRequest,
+) (*protos.SessionTerminateResponse, error) {
+
+	return nil, nil
+}
+
+// createN7Client creates a N7 oapi client and sets the OAuth2 client credentiatls for authorizing requests
+func createN7Client(cfg *PCFConfig) (*n7_client.ClientWithResponses, error) {
+	httpClient := &http.Client{}
+	tokenConfig := clientcredentials.Config{
+		ClientID:     cfg.ClientId,
+		ClientSecret: cfg.ClientSecret,
+		TokenURL:     cfg.TokenUrl,
+	}
+	tokenCtxt := context.WithValue(context.Background(), oauth2.HTTPClient, httpClient)
+	doer := tokenConfig.Client(tokenCtxt)
+
+	serverStr, err := GetServerStringFromUrl(cfg.ApiRoot)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create new N7 client object and assosiate it with oAuth2 HTTP client
+	client, err := n7_client.NewClientWithResponses(
+		serverStr,
+		n7_client.WithHTTPClient(doer),
+	)
+
+	return client, err
+}
+
+// Disable closes all existing pcf connections and disables
+// connection creation for the time specified in the request
+func (srv *CentralSessionController) Disable(
+	ctx context.Context,
+	req *fegprotos.DisableMessage,
+) (*orcprotos.Void, error) {
+	if req == nil {
+		return nil, fmt.Errorf("nil disable request")
+	}
+	// PCF Connections are stateless HTTP connections. Don't have to disable them.
+	if !srv.cfg.DisableN7 {
+		// No new requestes are made
+		srv.cfg.DisableN7 = true
+		disablePeriod := time.Duration(req.DisablePeriodSecs) * time.Second
+		time.AfterFunc(disablePeriod, func() { srv.cfg.DisableN7 = false })
+	}
+	return &orcprotos.Void{}, nil
+}
+
+// Enable enables pcf connection creation
+func (srv *CentralSessionController) Enable(
+	ctx context.Context,
+	void *orcprotos.Void,
+) (*orcprotos.Void, error) {
+	// PCF Connections are stateless HTTP connections.
+	srv.cfg.DisableN7 = false
+	return &orcprotos.Void{}, nil
+}
+
+// GetHealthStatus retrieves a health status object which contains the current
+// health of the service
+func (srv *CentralSessionController) GetHealthStatus(
+	ctx context.Context,
+	void *orcprotos.Void,
+) (*fegprotos.HealthStatus, error) {
+	currentMetrics, err := metrics.GetCurrentHealthMetrics()
+	if err != nil {
+		return &fegprotos.HealthStatus{
+			Health:        fegprotos.HealthStatus_UNHEALTHY,
+			HealthMessage: fmt.Sprintf("Error occurred while retrieving health metrics: %s", err),
```

Comment:
since this will be returned to the caller, it is better to return the service name 
```
Error occurred while retrieving health metrics on N7:
```